### ISSUE TITLE
[apex] ApexCRUDViolation doesn't report Database class DMLs, inline no-arg object instantiations and inline list initialization

### ISSUE THREAD

**Issue by jonathanwiesel** [pmd/pmd#3201]
**Affects PMD Version:** 
5.5.3 - 6.36.0

**Rule:**

[ApexCRUDViolation](https://pmd.github.io/latest/pmd_rules_apex_security.html#apexcrudviolation)

**Description:**

When using the `Database` class, inline list initialization or no-arg object initialization in the same statement as a DML call the ApexCRUDViolation rule fails to report it

**Code Sample demonstrating the issue:**

```
/**
 * @description X
 */
public with sharing class CRUDCheck {
    
    /**
     * @description X
     */
    public static void checkMe() {

        Account acc = new Account(Name = 'X');

        //these gets reported
        insert acc; 
        insert new Account(Name = 'X');
        

        // these doesn't get reported
        insert new Account();
        insert new Account[]{new Account()};
        insert new Account[]{new Account(Name = 'X')};
        Database.insert(acc);
        Database.insert(new Account());
        Database.insert(new Account(Name = 'X'));
    }
}

```

**Expected outcome:**

*   Is PMD missing to report a violation, where there should be one? -> false-negative

**Running PMD through:** *[CLI | Ant | Maven | Gradle | Designer | Other]*

Apex PMD VSCode Extension

**Comment by jonathanwiesel** [comment]
The `Database` class ones may refer to the same as #2628

**Comment by adangel** [comment]
See also #3378
